lib/handsoap/service.rb

```diff
@@ -96,9 +96,10 @@ class Service
     @@create_document_callback = nil
     @@log_header_callback = nil
     @@log_body_callback = nil
-    attr_reader :http_client
+    attr_reader :http_client, :cookie
     def initialize(ep)
       @http_client = nil
+      @cookie = nil
```

Comment:
Yeah the only problem is you get to the http body differently for `Curl` and `HTTPClient` (`:body_str` for one and `:content` for the other) so I ended up with this in `#document`
```
-        doc = Nokogiri::XML(@http_body)
+        http_body = if @http_response.respond_to?(:content)
+                      @http_response.content
+                    else
+                      @http_response.body_str
+                    end
+
+        doc = Nokogiri::XML(http_body)
```
which looks a little gross